### PR TITLE
fix warnings in ssr-tests-v9

### LINKED ISSUE #24184: [Bug]: ssr-tests-v9 produces warnings
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

N/A

### Bug Description

- Run `yarn workspace @fluentui/ssr-tests-v9 build`

## Actual Behavior

```
Warning: <storiesDefault /> is using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements.
    in storiesDefault
    in div
    in Context.Provider
    in TextDirectionProvider
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in FluentProvider
    in Context.Provider
    in SSRProvider
    in App
    in Context.Provider
    in RendererProvider
```

## Expected Behavior

No warnings

### Logs

```shell
^ see above
```

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by layershifter**
## Current Behavior

Warnings in console, see the issue (#24184).

## New Behavior

- No warnings
- New strict checks for directories where stories are located (as they are used a prefix in a component name)
- Stories from `react-dialog` were moved to follow the structure

## Related Issue(s)

Fixes #24184.

### CODE REVIEW

File: apps/ssr-tests-v9/src/utils/buildAssets.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ const commonOptions: BuildOptions = {
     '.jpg': 'dataurl',
     '.png': 'dataurl',
     '.svg': 'dataurl',
+    '.md': 'text',
```
Comment (layershifter) [inline apps/ssr-tests-v9/src/utils/buildAssets.ts:12]: Stories in `react-dialog` use imported markdown descriptions

File: apps/ssr-tests-v9/src/utils/getImportsFromIndexFile.ts [hunk 2]
```diff
@@ -8,15 +8,19 @@ type StoryImport = {
   path: string;
 };
 
-// TODO: Once will be solved, can be simplified https://github.com/microsoft/fluentui/issues/23393
```
Comment (layershifter) [inline apps/ssr-tests-v9/src/utils/getImportsFromIndexFile.ts:11]: This was completed, can be simplified

File: apps/ssr-tests-v9/src/utils/getImportsFromIndexFile.ts [hunk 3]
```diff
@@ -8,15 +8,19 @@ type StoryImport = {
...
-    return path.basename(path.dirname(filename));
+  const importPrefix = path.basename(path.dirname(filename));
+
+  if (importPrefix === 'stories' || importPrefix[0].toLowerCase() === importPrefix[0]) {
+    throw new Error(
+      [
+        `The file "${filename}" has an incorrect location a directory name.`,
+        `All stories should be placed in "/src/stories/ComponentName" directory.`,
+      ].join(' '),
+    );
```
Comment (layershifter) [inline apps/ssr-tests-v9/src/utils/getImportsFromIndexFile.ts:20]: A new check to assert that stories are not inside `stories` directory & directory name starts with a capital letter.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `190.59 kB`<br />`52.497 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `32.036 kB`<br />`10.577 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `83.94 kB`<br />`25.082 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0e1784cd430525e023a458564d232ac689e0f50a'>0e1784cd430525e023a458564d232ac689e0f50a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/Avatar-mount_base.html">1640</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/Avatar-mount.html">1617</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/Button-mount_base.html">1109</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/Button-mount.html">1068</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProvider-mount_base.html">2004</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProvider-mount.html">1886</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-mount_base.html">724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-mount.html">720</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">650</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">708</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/MakeStyles-mount_base.html">2315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/MakeStyles-mount.html">2272</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/SpinButton-mount_base.html">3404</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24186/perf-test-react-components/SpinButton-mount.html">3405</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ling1726 — APPROVED** [review]

**Pull request merged**